content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,380 @@
+                    className={`form-control`}
+                    placeholder="password"
+                    value={password}
+                    onChange={(e) => setPassword(e.target.value)}
+                  />
+               </div>
+               <div className={style.form__group}>
+                <p>Confirm Password</p>
+                <input
+                  name="confirm__pwd"
+                  id="pwd__conf"
+                  className={`form-control`}
+                  placeholder="enter your password again"
+                  value={pwdConfirm}
+                  onChange={(e) => setPwdConfrim(e.target.value)}
+                />
+                <button className={`btn-primary`}>Sign up</button>
+            </form>
+        </div>
+    </section>
+  );
+};
+
+export default withAlert()(SignUp);
+```
+
+Taking a look at the `handleSubmit` function, you’d notice how the arguments from the `signUpCheck` validator are being utilized. In the function, all conditions must be met before the form can be submitted.
+
+```js
+const validate = validateSignUp(email, password, pwdConfirm, alert ||" ");
+
+    if (validate) {
+      alert.success("You've signed up successfully. Proceed to login");
+    }
+};
+```
+
+- **Setting up the alert template**
+
+This step involves providing the `react-alert` template API to our application. Without this, the error messages wouldn’t be displayed on the webpage. Let’s start by taking a look at the setup below.
+
+We’d start by creating a component that’d serve as a container for the error messages. In the `utils` folder, we’d add all the logic of the alert-template.
+
+```js
+import React from "react";
+
+import { transitions, positions, Provider as AlertProvider } from "react-alert";
+
+const AlertTemplate = ({ message, options }) => {
+  return (
+    <div className={options.type === "success" ? "msg-success" : "msg-error"}>
+      <p>{message}</p>
+    </div>
+  );
+};
+
+// this component serves as the container
+// that holds/displays the error messages due to the validation
+// script that runs on all the app components that have an
+// input field
+const Message = ({ children }) => {
+  const options = {
+    position: positions.TOP_RIGHT,
+    timeout: 3500,
+    offset: "0px",
+    transition: transitions.SCALE,
+  };
+
+  return (
+    <AlertProvider template={AlertTemplate} {...options}>
+      {children}
+    </AlertProvider>
+  );
+};
+
+export default Message;
+```
+
+This snippet below shows a conditional statement but in a ternary syntax. What this translates to, is: when the `options.type` is “success”, add a CSS class of “msg-success”, if not, add a CSS class of “msg-error”.
+
+The “msg-success” gives the alert template/modal a green background while “msg-error” adds a red background to the template.
+
+Once that is complete, we’d go ahead and add the template as a parent element in the app component. This can be done by editing the content of `_app.js` file.
+
```

Comment:
```suggestion
Let's now go ahead and add the template as a parent element in the app component. This can be done by editing the content of `_app.js` file.

```